tensorflow/contrib/cmake/external/snappy.cmake

```diff
@@ -0,0 +1,47 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+include (ExternalProject)
+
+set(snappy_URL https://github.com/google/snappy.git)
+set(snappy_TAG "1.1.6")
+set(snappy_BUILD ${CMAKE_CURRENT_BINARY_DIR}/snappy/src/snappy)
+set(snappy_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/snappy/src/snappy)
+
+if(WIN32)
+    set(snappy_STATIC_LIBRARIES ${snappy_BUILD}/$(Configuration)/snappy.lib)
+endif()
```

Comment:
I used the same version of SNAPPY that was used in Bazel builds - 1.1.6. In that version Snappy's CMakeLists.txt explicitly builds shared library:

```
# Define the main library.
ADD_LIBRARY(snappy SHARED
```

After 1.1.6 release in master branch this behavior was changed:
`add_library(snappy "")`

I suppose it'd be right to use the recent commit on master instead of 1.1.6 tag and build Snappy as a static library. What do you think?